torch/csrc/dynamo/guards.cpp

```diff
@@ -292,12 +294,25 @@ PyObject* TensorGuards_check(TensorGuards* self, PyObject* args) {
   }
 
   LocalState state;
-
+  // Note - all the tensors that make it to guards must be unique. Dynamo
+  // builder handles guarding for positive aliases (X is Y). However, we do not
+  // create guards for negative alias (X is not Y) as that is an N^2
+  // relationship. Instead, we rely on the uniqueness upstream to verify, at
+  // check_fn time (this function).
+  std::set<PyObject*> unique_tensors;
```

Comment:
Consider using flat_hash_map instead, std::set is famously slow

Reply:
Sounds good!